src/TagsInput/TagsInput.js

```diff
@@ -11,156 +16,159 @@ import InputField from './components/InputField';
+    this.setState({ value: '' }, onClearPress);
+    this.handleOnFocus();
+  };
+
   render() {
-    const { selected, fontSize, label, disabled } = this.props;
-    const { value, inputWidth, isFocus } = this.state;
-    const isButtonDisabled = !value.length ?? disabled;
+    const { disabled, fontSize, label, onKeyPress, tags } = this.props;
+    const { isFocus, value } = this.state;
 
-    const dynamicStyle = StyleSheet.create({
+    const isButtonDisabled = !value ?? disabled;
+
+    const dynamicStyle = {
       deleteButton: {
-        opacity: value.length > 0 ? 1 : 0,
+        opacity: value ? 1 : 0,
       },
       label: {
         fontSize,
       },
       border: {
-        web: {
-          boxShadow: isFocus
-            ? `${defaultTokens.borderColorInputFocus} 0 0 0 2px inset`
-            : `${defaultTokens.borderColorInput} 0 0 0 1px inset`,
-        },
+        ...boxShadow(isFocus),
       },
-    });
+    };
 
     return (
-      <View style={[styles.container, dynamicStyle.border, dynamicStyle.label]}>
-        <Text weight="bold" style={styles.label}>
-          {label}
-        </Text>
-        <View style={styles.fieldContainer} onLayout={this.setContainerWidth}>
-          <TagsContainer
-            minWidth={TAGS_MIN_WIDTH}
-            maxWidth={this.subtractFromContainerWidth(inputWidth)}
-            tags={selected}
-            fontSize={fontSize}
-          />
-          <InputField
-            onFocus={this.handleOnFocus}
-            onBlur={this.handleOnBlur}
-            ref={this.inputRef}
-            fontSize={fontSize}
-            disabled={disabled}
-            minWidth={INPUT_MIN_WIDTH}
-            maxWidth={this.subtractFromContainerWidth(TAGS_MIN_WIDTH)}
-            onLayout={this.setInputWidth}
-            value={value}
-            placeholder={this.getPlaceholder()}
-            onChangeText={this.handleChange}
-            autoFocus={isFocus}
+      <TouchableWithoutFeedback onPress={this.handleOnFocus}>
+        <View style={[styles.container, dynamicStyle.border]}>
+          {label && (
+            <Text weight="bold" style={[styles.label, dynamicStyle.label]}>
+              {label}
+            </Text>
+          )}
+          <ScrollView
+            scrollEventThrottle={3}
+            showsHorizontalScrollIndicator={false}
+            horizontal
+            style={styles.fieldContainer}
+            ref={this.scrollRef}
+          >
+            <TagsContainer tags={tags} fontSize={fontSize} />
+            <InputField
+              onKeyPress={onKeyPress}
+              onFocus={this.handleOnFocus}
+              onBlur={this.handleOnBlur}
+              ref={this.inputRef}
+              fontSize={fontSize}
+              disabled={disabled}
+              value={value}
+              placeholder={this.getPlaceholder()}
+              onChangeText={this.handleChange}
+            />
+          </ScrollView>
+          <DeleteButton
```

Comment:
I'm not sure if this button is OK from the UX perspective. It's quite unexpected that you can see this button only when you have something typed on keyboard and you can't delete with this button selected places from the list. 

I think it should works like that you can delete any place with this button (typed or selected) and it should be visible all the time except blank input. In this case also icon should be different, something like that: https://image.shutterstock.com/image-vector/delete-backspace-keyboard-key-line-450w-273565205.jpg

Current icon with cross would be also fine if we could delete all the content of the input with it. That is also approach which use Kiwi.com app. Maybe it would be the easiest change.

It's nothing serious for this PR, but we should keep it on mind in the future when we will improve UX and design of this component.

Reply:
done

Reply:
Now it's much better, thanks!